docs/api-reference/xviz-validator.md

```diff
@@ -14,15 +14,15 @@ All validation methods throw if the object fails to validate.
 
 ##### validateMetadata(data)
 
-Validate `metadata` information.
+Validate `METADATA` information.
 
 Parameters:
 
 - `data` (Object)
 
 ##### validateStateUpdate(data)
 
-Validate `state_update` message.
+Validate `STATE_UPDATE` message.
```

Comment:
These I am a little unsure on whether to be `STATE_UPDATE` or shift them to being `StateUpdate` to match JavaScript/C++/Go type naming conventions since they are really types.  If we change them to `StateUpdate` it will require some additional work because they are encoded into the file system.

Reply:
So this relates to the wire protocol/message type distinction i mentioned in an earlier comment.  In this context we are talking about the **message**.  So you are thinking we describe the message type as **StateUpdate**, but then it is represented in the wire protocol as **xviz/state_update**?  Or are you thinking of changing that as well?

Again, I have been treating all 3 context, the wire protocol, the identifier, and the name as all being in alignment for simplicity but I can see that is not absolutely necessary.

However, if we have carried this to the file system, your assuming we would keep them lower-case but remove the underscore right, so the file`state_update` => `stateupdate`.  filename casing has caused problems cross platforms at times.

Reply:
Yeah thinking about what we have:
 - In the envelope we have `namespace/high_level_type` like `xviz/state_update`
 - in spec we have types like `StateUpdate` in protobuf (or basically any language under the sun), some types in the spec are the high level types we send over the wire
 - in the filesystem we have `examples/state_update/foo`

My idea with this change was only to fix the enums and leave all of the above alone, although it would be nice to get everything straightened out.

Reply:
Ok.  So I will just focus on enums.  As long as the I know the relationship and can document the convention I think that is good enough.  I'll peek at the *type* issue and how that effects things.

One area that is a bit awkward is the handling of `namespace/high_level_type` through the code.  Right now we mostly do the right thing but it is not very clear why.  By this i mean we use the **high_level_type** in some places, and the full **namespace/high_level_type** in others.

Ideally we can use the former internally once we have parsed and validated the message as being an **xviz** namespace.  But only message validation should concern itself with the latter.